aten/src/ATen/TensorOptions.h

```diff
@@ -17,10 +17,29 @@ namespace at {
 
 /// A class to encapsulate construction axes of a `Tensor`.
 /// `TensorOptions` is a virtual class to enable overriding of certain methods
```

Comment:
I'd add some comments about the motivation of 'TensorOptions'. Why is this important?

Reply:
added